prompt/prompt-executor/prompt-executor-clients/prompt-executor-google-client/src/jvmTest/kotlin/ai/koog/prompt/executor/clients/google/GoogleLLMClientTest.kt

```diff
@@ -185,23 +190,139 @@ class GoogleLLMClientTest {
             tools = listOf(tool)
         )
 
-        assertNotNull(request.tools)
-        val tools = request.tools!!
+        request.tools.shouldNotBeNull()
+        val tools = request.tools
         val functionDeclarations = tools.first().functionDeclarations!!
         val parameters = functionDeclarations.first().parameters!!
         val properties = parameters["properties"]?.jsonObject!!
-        assertNotNull(properties)
         val complexValue = properties["complexValue"]?.jsonObject!!
-        assertNotNull(complexValue)
 
         val anyOf = complexValue["anyOf"]?.jsonArray
-        assertNotNull(anyOf)
-        assertEquals(3, anyOf.size, "anyOf should have 3 options")
+        anyOf.shouldNotBeNull()
+        anyOf.size shouldBe 3
 
         // Verify the types
         val types = anyOf.map { it.jsonObject["type"]?.jsonPrimitive?.content }
-        assertTrue(types.contains("string"), "Should contain string type")
-        assertTrue(types.contains("number"), "Should contain number type")
-        assertTrue(types.contains("null"), "Should contain null type")
+        types shouldContain "string"
+        types shouldContain "number"
+        types shouldContain "null"
+    }
+
+    @Test
+    fun `createGoogleRequest should map GoogleParams to generationConfig`() {
+        val client = GoogleLLMClient(apiKey = "apiKey")
+        val model = GoogleModels.Gemini2_5Pro
+
+        val params = GoogleParams(
+            temperature = 0.4,
+            maxTokens = 1024,
+            numberOfChoices = 2,
+            topP = 0.8,
+            topK = 10,
+            thinkingConfig = GoogleThinkingConfig(
+                includeThoughts = true,
+                thinkingBudget = 99
+            ),
+            additionalProperties = mapOf("custom" to JsonPrimitive("v"))
+        )
+
+        val request = client.createGoogleRequest(
+            prompt = Prompt(messages = emptyList(), id = "id", params = params),
+            model = model,
+            tools = emptyList()
+        )
+
+        val gen = request.generationConfig!!
+        gen.maxOutputTokens shouldBe 1024
+        gen.temperature shouldBe 0.4
+        gen.candidateCount shouldBe 2
+        gen.topP shouldBe 0.8
+        gen.topK shouldBe 10
+        gen.thinkingConfig?.includeThoughts shouldBe true
+        gen.thinkingConfig?.thinkingBudget shouldBe 99
+        gen.additionalProperties.shouldNotBeNull()
+        gen.additionalProperties["custom"]?.jsonPrimitive?.content shouldBe "v"
```

Comment:
with kotest you may assert non-null value inside lambda with receiver:
```suggestion
        request.generationConfig shouldNotBeNull { // this=generationConfig
            maxOutputTokens shouldBe 1024
            temperature shouldBe 0.4
            candidateCount shouldBe 2
            topP shouldBe 0.8
            topK shouldBe 10
            thinkingConfig?.includeThoughts shouldBe true
            thinkingConfig?.thinkingBudget shouldBe 99
            additionalProperties.shouldNotBeNull()
            additionalProperties["custom"]?.jsonPrimitive?.content shouldBe "v"
        }
        
```